platform/config-core/src/main/java/org/motechproject/config/core/filestore/ConfigLocationFileStore.java

```diff
@@ -34,12 +37,41 @@ public ConfigLocationFileStore(PropertiesConfiguration propertiesConfiguration)
      *
      * @return the list of configuration locations
      */
-    public Iterable<ConfigLocation> getAll() {
-        List<String> configLocations = loadAll();
-
+    public Iterable<ConfigLocation> getAllConfigLocations() {
+        List<String> locations = getLocations();
+        List<String> configLocations = new ArrayList<>();
+        for (String location : locations) {
+            configLocations.add(new String(location + "/config/"));
```

Comment:
As mentioned, I would expect the config to be appended during the save of the location. Moreover, we should be a bit more clever about file separators, pretty sure this can go wrong on Windows or when the value of the location var ends with a slash already.